extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -152,56 +152,35 @@ export default class DebugPlugin extends BaseTelemetryPlugin {
           debugBinContainer.className = `${prefix}-debug-bin-container`;
           debugBinParent = new DebugBinParent(debugBinContainer, [], 0, prefix);
 
-          // 4. grab all relevant functions that need to be instrumented and store them in arrays
-          const diagLog = _self.diagLog();
 
+          // 5. instrument functions
+          const diagLog = _self.diagLog();
           debugBins = {};
-          const propertiesProtoFns: string[] = [];
-          const analyticsProtoFns: string[] = [];
-          const ajaxProtoFns: string[] = [];
-          const channelProtoFns: string[] = [];
-          const diagLogProtoFns: string[] = [];
-          for (const [ext, protoFns] of [
-            [analyticsExt, analyticsProtoFns],
-            [propertiesExt, propertiesProtoFns],
-            [ajaxDependencyExt, ajaxProtoFns],
-            [channelPluginExt, channelProtoFns],
-            [diagLog, diagLogProtoFns]
-          ] as any[]) {
-            for (const key of CoreUtils.objKeys(ext['__proto__'])) {
-              if (key.substring(0, 1) === '_') { continue; }
-              if (CoreUtils.isFunction(ext[key])) {
-                protoFns.push(key);
+          for (let i = 0; i < trackers.length; i++) {
+            let tracker = trackers[i];
+            for (const target of [propertiesExt, analyticsExt, ajaxDependencyExt, channelPluginExt, diagLog]) {
+              let val = InstrumentFunc(target, tracker, {
+                req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
+                rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
+              }, true);
+
+              if (!val) {
```

Comment:
As your passing true to the checkPrototype argument (last one) above you this should be dead code as the above will already check and hook the prototype.